cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/transform/ScriptLookupTable.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.etl.transform;
+
+import sun.org.mozilla.javascript.internal.NativeObject;
```

Comment:
An alternate way is to have a function on the JS side that can convert a `StructuredRecord` into JS object. A slower approach is to leverage the existing `StructuredRecordStringConverter` by first converting a `StructuredRecord` into a JSON string, followed by calling `JSON.parse()` on the JS side. Something like this:

```java
    ScriptEngineManager manager = new ScriptEngineManager();
    ScriptEngine engine = manager.getEngineByName("JavaScript");
    engine.eval("function foo(lookup) { println(lookup.get('name').id); }");
    engine.eval("function getJson() { return JSON; }");

    Invocable invocable = (Invocable) engine;
    invocable.invokeFunction("foo", new LookupImpl(invocable, invocable.invokeFunction("getJson")));

...

  public static final class LookupImpl implements Lookup {

    private final Invocable invocable;
    private final Object json;

    public LookupImpl(Invocable invocable, Object json) {
      this.invocable = invocable;
      this.json = json;
    }

    @Override
    public Object get(String key) {
      StructuredRecord record = StructuredRecord.builder(
        Schema.recordOf("Record", Schema.Field.of("id", Schema.of(Schema.Type.STRING))))
        .set("id", key)
        .build();
      try {
        return invocable.invokeMethod(json, "parse", StructuredRecordStringConverter.toJsonString(record));
      } catch (Exception e) {
        throw Throwables.propagate(e);
      }
    }
```